runtime/restart/restart.go

```diff
@@ -44,12 +47,88 @@ const (
 	// LogURILabel sets the restart log uri label for a container
 	LogURILabel = "containerd.io/restart.loguri"
 
+	// PolicyLabel sets the restart policy label for a container
+	PolicyLabel = "containerd.io/restart.policy"
+	// CountLabel sets the restart count label for a container
+	CountLabel = "containerd.io/restart.count"
+
 	// LogPathLabel sets the restart log path label for a container
 	//
 	// Deprecated(in release 1.5): use LogURILabel
 	LogPathLabel = "containerd.io/restart.logpath"
 )
 
+// Policy represents the restart policies of a container.
+type Policy struct {
+	name              string
+	maximumRetryCount int
+}
+
+// NewPolicy creates a restart policy with the specified name.
+// supports the following restart policies:
+// - no, Do not restart the container.
+// - always, Always restart the container regardless of the exit status.
+// - on-failure[:max-retries], Restart only if the container exits with a non-zero exit status.
+// - unless-stopped, Always restart the container unless it is stopped.
+func NewPolicy(policy string) (*Policy, error) {
```

Comment:
Can we expose the list of the implemented policies as the plugin meta data?
https://github.com/containerd/containerd/blob/d394e00c7eb9ed22b842f0d75c20e9353dccf488/plugin/context.go#L66

Reply:
Can we set `Meta.Capabilities = {“no”, “always”, “on-failure”, “unless-stopped”}` ?

https://github.com/containerd/containerd/blob/82bce322da3fc41e7e7ffc3dcfc85e66fe54a75e/api/services/introspection/v1/introspection.proto#L62-L73

Reply:
SGTM.